src/utils/errors.js

```diff
@@ -1,10 +1,106 @@
 export class AppError extends Error {
-  constructor(message, statusCode) {
+  constructor(message, statusCode = 500) {
     super(message);
+    this.name = this.constructor.name;
     this.statusCode = statusCode;
-    this.status = `${statusCode}`.startsWith('4') ? 'fail' : 'error';
-    this.isOperational = true;
-
     Error.captureStackTrace(this, this.constructor);
   }
-}
\ No newline at end of file
+}
+
+// 404 - Not Found Error - відсутні ресурси
+export class NotFoundError extends AppError {
+  constructor(message = 'Resource not found') {
+    super(message, 404);
+  }
+}
+
+// 400 - Validation Error - неправильні вхідні дані
+export class ValidationError extends AppError {
```

Comment:
extends `ClientError`

also for validation errors you can use 422 HTTP status.

Reply:
Also you can include for example the array of error messages to this class in constructor. and then use it to include to HTTP response body.